contracts/src/LightClient.sol

```diff
@@ -361,42 +362,46 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     {
         uint256 updatesCount = stateHistoryCommitments.length;
 
-        // Handling Edge Cases
-        // Edgecase 1: The block is in the future or
-        // before HotShot was live, allow for at least two updates before considering HotShot live
-        if (blockNumber > block.number || updatesCount < 3) {
+        // Edge Case Handling:
+        // 1. Provided block number is greater than the current block (invalid)
+        // 2. No updates have occurred (i.e., state history is empty)
+        // 3. Provided block number is earlier than the first recorded state update
+        // the stateHistoryFirstIndex is used to check for the first nonZero element
+        if (
+            blockNumber > block.number || updatesCount == 0
+                || blockNumber < stateHistoryCommitments[stateHistoryFirstIndex].l1BlockHeight
+        ) {
             revert InsufficientSnapshotHistory();
         }
 
-        uint256 prevBlock;
-        bool prevUpdateFound;
+        uint256 eligibleBlock; // the eligible block is <= blockNumber
```

Comment:
I think something like "lastStateUpdateBeforeBlockNumber" might be clearer?

Reply:
well the last state update can also be the same block number as the one being assessed. So I wanted to remove terminology like `prev` or `before` so that it's not misleading when reading the code

Reply:
How about `eligibleStateUpdateBlockNumber` then?

Reply:
udpated here https://github.com/EspressoSystems/espresso-sequencer/commit/3d5ae7aefea737d7611d26f78a02c08144029481#diff-d85261c27bf48fe0ec00e8cc9bb3ca101d708ec68c85968531b2e2cf4f9ed292R377